framework/wazuh/core/cluster/cluster.py

```diff
@@ -421,6 +425,16 @@ def split_on_condition(seq, condition):
                                             lambda x: cluster_items[check_files[x]['cluster_item_key']]['extra_valid'])
     extra_files = {key: check_files[key] for key in extra}
     extra_valid_files = {key: check_files[key] for key in extra_valid}
+    if extra_valid_files:
+        # Check if extra-valid agent-groups files correspond to existing agents.
+        try:
+            agent_groups = [os.path.basename(file) for file in extra_valid_files if file.startswith(agent_groups_path)]
+            db_agents = Agent.get_agents_overview(select=['id'], limit=None, filters={'id': agent_groups})['items']
+            db_agents = set(map(itemgetter('id'), db_agents))
```

Comment:
Is it really necessary to use the `itemgetter` library? We select only the id

Reply:
In this specific case, I think it is. The response of `get_agnets_overview` is like this: [{'id': '001'}, {'id': '002'}, {'id': '003'}], but we need a set containing only the ID, not a dict for each one.

Reply:
We need to test this: `{d['id'] for d in list}` vs itemgetter for this case